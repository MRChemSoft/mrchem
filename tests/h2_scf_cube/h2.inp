# vim:syntax=sh:

world_prec = 1.0e-3               # Overall relative precision
world_size = 5                    # Size of simulation box 2^n

MPI {
  numerically_exact = true        # Guarantee identical results in MPI
}

Basis {
  order = 7                       # Polynomial order
  type = Legendre                 # Polynomial type (Legendre or Interpolating)
}


Molecule {
$coords
H   0.000000    0.000000    0.672828
H   0.000000    0.000000   -0.672828
$end
}

WaveFunction {
  method = HF                     # Wave function method (HF or DFT)
  restricted = true
}

SCF {
  kain = 3                        # Length of KAIN iterative history
  max_iter = 5
  orbital_thrs = 1.0e-2           # Convergence threshold in orbital residual
  guess_type = cube             # Type of initial guess: none, mw, gto, cube
}

