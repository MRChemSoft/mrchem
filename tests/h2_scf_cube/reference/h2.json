{
  "input": {
    "constants": {
      "angstrom2bohrs": 1.8897261246257702,
      "dipmom_au2debye": 2.5417464739297717,
      "electron_g_factor": -2.00231930436256,
      "fine_structure_constant": 0.0072973525693,
      "hartree2ev": 27.211386245988,
      "hartree2kcalmol": 627.5094740630558,
      "hartree2kjmol": 2625.4996394798254,
      "hartree2simagnetizability": 78.9451185,
      "hartree2wavenumbers": 219474.6313632,
      "light_speed": 137.035999084
    },
    "molecule": {
      "charge": 0,
      "coords": [
        {
          "atom": "h",
          "xyz": [
            0.0,
            0.0,
            0.672828
          ]
        },
        {
          "atom": "h",
          "xyz": [
            0.0,
            0.0,
            -0.672828
          ]
        }
      ],
      "multiplicity": 1
    },
    "mpi": {
      "bank_size": -1,
      "numerically_exact": true,
      "shared_memory_size": 10000
    },
    "mra": {
      "basis_order": 7,
      "basis_type": "legendre",
      "boxes": [
        2,
        2,
        2
      ],
      "corner": [
        -1,
        -1,
        -1
      ],
      "max_scale": 20,
      "min_scale": -4
    },
    "printer": {
      "file_name": "h2",
      "print_constants": false,
      "print_level": 0,
      "print_mpi": false,
      "print_prec": 6,
      "print_width": 75
    },
    "rsp_calculations": {},
    "scf_calculation": {
      "fock_operator": {
        "coulomb_operator": {
          "poisson_prec": 0.001,
          "shared_memory": false
        },
        "exchange_operator": {
          "exchange_prec": -1.0,
          "poisson_prec": 0.001
        },
        "kinetic_operator": {
          "derivative": "abgv_55"
        },
        "nuclear_operator": {
          "proj_prec": 0.001,
          "shared_memory": false,
          "smooth_prec": 0.001
        }
      },
      "initial_guess": {
        "environment": "None",
        "external_field": "None",
        "file_CUBE_a": "cube_vectors/CUBE_a_vector.json",
        "file_CUBE_b": "cube_vectors/CUBE_b_vector.json",
        "file_CUBE_p": "cube_vectors/CUBE_p_vector.json",
        "file_basis": "initial_guess/mrchem.bas",
        "file_chk": "checkpoint/phi_scf",
        "file_gto_a": "initial_guess/mrchem.moa",
        "file_gto_b": "initial_guess/mrchem.mob",
        "file_gto_p": "initial_guess/mrchem.mop",
        "file_phi_a": "initial_guess/phi_a_scf",
        "file_phi_b": "initial_guess/phi_b_scf",
        "file_phi_p": "initial_guess/phi_p_scf",
        "localize": false,
        "method": "Hartree-Fock",
        "prec": 0.001,
        "relativity": "None",
        "restricted": true,
        "screen": 12.0,
        "type": "cube",
        "zeta": 0
      },
      "properties": {
        "dipole_moment": {
          "dip-1": {
            "operator": "h_e_dip",
            "precision": 0.001,
            "r_O": [
              0.0,
              0.0,
              0.0
            ]
          }
        }
      },
      "scf_solver": {
        "checkpoint": false,
        "energy_thrs": -1.0,
        "environment": "None",
        "external_field": "None",
        "file_chk": "checkpoint/phi_scf",
        "final_prec": 0.001,
        "helmholtz_prec": -1.0,
        "kain": 3,
        "localize": false,
        "max_iter": 5,
        "method": "Hartree-Fock",
        "orbital_thrs": 0.01,
        "relativity": "None",
        "rotation": 0,
        "start_prec": 0.001
      }
    },
    "schema_name": "mrchem_input",
    "schema_version": 1
  },
  "output": {
    "properties": {
      "center_of_mass": [
        0.0,
        0.0,
        0.0
      ],
      "charge": 0,
      "dipole_moment": {
        "dip-1": {
          "magnitude": 0.00012113100321773263,
          "r_O": [
            0.0,
            0.0,
            0.0
          ],
          "vector": [
            8.562070410903622e-05,
            8.562070412136231e-05,
            -3.3030280303609777e-06
          ],
          "vector_el": [
            8.562070345631334e-05,
            8.562070345614548e-05,
            -3.30302803191529e-06
          ],
          "vector_nuc": [
            0.0,
            0.0,
            0.0
          ]
        }
      },
      "geometry": [
        {
          "symbol": "H",
          "xyz": [
            0.0,
            0.0,
            0.672828
          ]
        },
        {
          "symbol": "H",
          "xyz": [
            0.0,
            0.0,
            -0.672828
          ]
        }
      ],
      "multiplicity": 1,
      "orbital_energies": {
        "energy": [
          -0.604191218847401
        ],
        "occupation": [
          2.0
        ],
        "spin": [
          "p"
        ],
        "sum_occupied": -1.208382437694802
      },
      "scf_energy": {
        "E_ee": 1.3364093910495527,
        "E_eext": 0.0,
        "E_el": -1.8765896196296339,
        "E_en": -3.6991785024841493,
        "E_kin": 1.1543817009196817,
        "E_next": 0.0,
        "E_nn": 0.7431319742935788,
        "E_nuc": 0.7431319742935788,
        "E_tot": -1.1334576453360552,
        "E_x": -0.6682022091147191,
        "E_xc": 0.0,
        "Er_el": 0.0,
        "Er_nuc": 0.0,
        "Er_tot": 0.0
      }
    },
    "provenance": {
      "creator": "MRChem",
      "mpi_processes": 1,
      "nthreads": 8,
      "routine": "mrchem.x",
      "total_cores": 8,
      "version": "1.2.0-alpha"
    },
    "rsp_calculations": null,
    "scf_calculation": {
      "initial_energy": {
        "E_ee": 1.358939683386661,
        "E_eext": 0.0,
        "E_el": -1.8297495695126602,
        "E_en": -3.7448603981034205,
        "E_kin": 1.2356382042157266,
        "E_next": 0.0,
        "E_nn": 0.7431319742935788,
        "E_nuc": 0.7431319742935788,
        "E_tot": -1.0866175952190815,
        "E_x": -0.6794670590116274,
        "E_xc": 0.0,
        "Er_el": 0.0,
        "Er_nuc": 0.0,
        "Er_tot": 0.0
      },
      "scf_solver": {
        "converged": true,
        "cycles": [
          {
            "energy_terms": {
              "E_ee": 1.332826723172202,
              "E_eext": 0.0,
              "E_el": -1.8760020038042002,
              "E_en": -3.689442972994439,
              "E_kin": 1.1470250154088903,
              "E_next": 0.0,
              "E_nn": 0.7431319742935788,
              "E_nuc": 0.7431319742935788,
              "E_tot": -1.1328700295106215,
              "E_x": -0.6664107693908535,
              "E_xc": 0.0,
              "Er_el": 0.0,
              "Er_nuc": 0.0,
              "Er_tot": 0.0
            },
            "energy_total": -1.1328700295106215,
            "energy_update": 0.04625243429154002,
            "mo_residual": 0.04866549592087212,
            "wall_time": 0.746595584
          },
          {
            "energy_terms": {
              "E_ee": 1.3364093910495527,
              "E_eext": 0.0,
              "E_el": -1.8765896196296339,
              "E_en": -3.6991785024841493,
              "E_kin": 1.1543817009196817,
              "E_next": 0.0,
              "E_nn": 0.7431319742935788,
              "E_nuc": 0.7431319742935788,
              "E_tot": -1.1334576453360552,
              "E_x": -0.6682022091147191,
              "E_xc": 0.0,
              "Er_el": 0.0,
              "Er_nuc": 0.0,
              "Er_tot": 0.0
            },
            "energy_total": -1.1334576453360552,
            "energy_update": 0.0005876158254336517,
            "mo_residual": 0.009310882808245153,
            "wall_time": 0.704478375
          }
        ],
        "wall_time": 1.451236375
      },
      "success": true
    },
    "schema_name": "mrchem_output",
    "schema_version": 1,
    "success": true
  }
}
