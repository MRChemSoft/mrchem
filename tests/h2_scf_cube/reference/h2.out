                                                                           
                                                                           
***************************************************************************
***                                                                     ***
***                                                                     ***
***          __  __ ____   ____ _                                       ***
***         |  \/  |  _ \ / ___| |__   ___ _ __ ___                     ***
***         | |\/| | |_) | |   | '_ \ / _ \ '_ ` _ \                    ***
***         | |  | |  _ <| |___| | | |  __/ | | | | |                   ***
***         |_|  |_|_| \_\\____|_| |_|\___|_| |_| |_|                   ***
***                                                                     ***
***         VERSION            1.2.0-alpha                              ***
***                                                                     ***
***         Git branch         refactor_cubeparser                      ***
***         Git commit hash    249ff05f717fb91ac344-dirty               ***
***         Git commit author  ggerez                                   ***
***         Git commit date    Tue Oct 25 12:05:49 2022 +0200           ***
***                                                                     ***
***         Contact: luca.frediani@uit.no                               ***
***                                                                     ***
***         Radovan Bast            Magnar Bjorgve                      ***
***         Roberto Di Remigio      Antoine Durdek                      ***
***         Luca Frediani           Gabriel Gerez                       ***
***         Stig Rune Jensen        Jonas Juselius                      ***
***         Rune Monstad            Peter Wind                          ***
***                                                                     ***
***************************************************************************

---------------------------------------------------------------------------

 MPI processes           :       (no bank)                               1
 OpenMP threads          :                                               8
 Total cores             :                                               8
                                                                           
---------------------------------------------------------------------------

XCFun DFT library Copyright 2009-2020 Ulf Ekstrom and contributors.
See http://dftlibs.org/xcfun/ for more information.

This is free software; see the source code for copying conditions.
There is ABSOLUTELY NO WARRANTY; not even for MERCHANTABILITY or
FITNESS FOR A PARTICULAR PURPOSE. For details see the documentation.
Scientific users of this library should cite
U. Ekstrom, L. Visscher, R. Bast, A. J. Thorvaldsen and K. Ruud;
J.Chem.Theor.Comp. 2010, DOI: 10.1021/ct100117s

---------------------------------------------------------------------------

 MRCPP version         : 1.4.1
 Git branch            : HEAD
 Git commit hash       : 75d41879b1908a94a452
 Git commit author     : Stig Rune Jensen
 Git commit date       : Thu Jan 6 11:38:53 2022 +0100

 Linear algebra        : EIGEN v3.4.0
 Parallelization       : OpenMP (8 threads)

---------------------------------------------------------------------------


                                                                           
===========================================================================
                         MultiResolution Analysis
---------------------------------------------------------------------------
 polynomial order      : 7
 polynomial type       : Legendre
---------------------------------------------------------------------------
 total boxes           : 8
 boxes                 : [          2           2           2 ]
 unit lengths          : [   16.00000    16.00000    16.00000 ]
 scaling factor        : [    1.00000     1.00000     1.00000 ]
 lower bounds          : [  -16.00000   -16.00000   -16.00000 ]
 upper bounds          : [   16.00000    16.00000    16.00000 ]
 total length          : [   32.00000    32.00000    32.00000 ]
===========================================================================


                                                                           
***************************************************************************
***                                                                     ***
***                        Initializing Molecule                        ***
***                                                                     ***
***************************************************************************
                                                                           
                                                                           
===========================================================================
                                 Molecule
---------------------------------------------------------------------------
 Charge                  :                                               0
 Multiplicity            :                                               1
---------------------------------------------------------------------------
    N    Atom            :               x               y               z
---------------------------------------------------------------------------
    0       H            :        0.000000        0.000000        0.672828
    1       H            :        0.000000        0.000000       -0.672828
---------------------------------------------------------------------------
 Center of mass          :        0.000000        0.000000        0.000000
===========================================================================


                                                                           
***************************************************************************
***                                                                     ***
***                Computing Initial Guess Wavefunction                 ***
***                                                                     ***
***************************************************************************
                                                                           
                                                                           
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 Calculation             : Compute initial orbitals
 Method                  : Project cube file molecular orbitals
 Precision               : 1.00000e-03
 Restricted              : True
 MO file                 : cube_vectors/CUBE_p_vector.json
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~


===========================================================================
                            Molecular Orbitals
---------------------------------------------------------------------------
 Alpha electrons         :                                               1
 Beta electrons          :                                               1
 Total electrons         :                                               2
---------------------------------------------------------------------------
    n  Occ Spin          :                                            Norm
---------------------------------------------------------------------------
    0    2    p          :                              1.000000000000e+00
===========================================================================


~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 Calculation             : Compute initial energy
 Method                  : Hartree-Fock
 Relativity              : None
 Environment             : None
 External fields         : None
 Precision               : 1.00000e-03
 Localization            : Off
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~


===========================================================================
                        Molecular Energy (initial)
---------------------------------------------------------------------------
 Kinetic energy          :            (au)                  1.235638204216
 E-N energy              :            (au)                 -3.744860398103
 Coulomb energy          :            (au)                  1.358939683387
 Exchange energy         :            (au)                 -0.679467059012
 X-C energy              :            (au)                  0.000000000000
 N-N energy              :            (au)                  0.743131974294
---------------------------------------------------------------------------
 Electronic energy       :            (au)                 -1.829749569513
 Nuclear energy          :            (au)                  0.743131974294
---------------------------------------------------------------------------
 Total energy            :            (au)             -1.086617595219e+00
                         :      (kcal/mol)             -6.818628356836e+02
                         :        (kJ/mol)             -2.852914104500e+03
                         :            (eV)             -2.956837108519e+01
===========================================================================


===========================================================================
                        Orbital Energies (initial)
---------------------------------------------------------------------------
    n  Occ Spin          :                                         Epsilon
---------------------------------------------------------------------------
    0    2    p          :            (au)                 -0.575138472569
---------------------------------------------------------------------------
 Sum occupied            :            (au)                 -1.150276945138
===========================================================================


                                                                           
***************************************************************************
***                                                                     ***
***                 Computing Ground State Wavefunction                 ***
***                                                                     ***
***************************************************************************
                                                                           
                                                                           
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 Calculation             : Optimize ground state orbitals
 Method                  : Hartree-Fock
 Relativity              : None
 Environment             : None
 External fields         : None
 Checkpointing           : Off
 Max iterations          : 5
 KAIN solver             : 3
 Localization            : Off
 Diagonalization         : First two iterations
 Start precision         : 1.00000e-03
 Final precision         : 1.00000e-03
 Helmholtz precision     : Dynamic
 Energy threshold        : Off
 Orbital threshold       : 1.00000e-02
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~


===========================================================================
 Iter           MO residual             Total energy                Update
---------------------------------------------------------------------------
    0          1.000000e+00          -1.086617595219         -1.086618e+00
    1          4.866550e-02          -1.132870029511         -4.625243e-02
    2          9.310883e-03          -1.133457645336         -5.876158e-04
---------------------------------------------------------------------------
                      SCF converged in 2 iterations!
===========================================================================


                                                                           
***************************************************************************
***                                                                     ***
***                    Printing Molecular Properties                    ***
***                                                                     ***
***************************************************************************
                                                                           
                                                                           
===========================================================================
                                 Molecule
---------------------------------------------------------------------------
 Charge                  :                                               0
 Multiplicity            :                                               1
---------------------------------------------------------------------------
    N    Atom            :               x               y               z
---------------------------------------------------------------------------
    0       H            :        0.000000        0.000000        0.672828
    1       H            :        0.000000        0.000000       -0.672828
---------------------------------------------------------------------------
 Center of mass          :        0.000000        0.000000        0.000000
===========================================================================


===========================================================================
                         Molecular Energy (final)
---------------------------------------------------------------------------
 Kinetic energy          :            (au)                  1.154381700920
 E-N energy              :            (au)                 -3.699178502484
 Coulomb energy          :            (au)                  1.336409391050
 Exchange energy         :            (au)                 -0.668202209115
 X-C energy              :            (au)                  0.000000000000
 N-N energy              :            (au)                  0.743131974294
---------------------------------------------------------------------------
 Electronic energy       :            (au)                 -1.876589619630
 Nuclear energy          :            (au)                  0.743131974294
---------------------------------------------------------------------------
 Total energy            :            (au)             -1.133457645336e+00
                         :      (kcal/mol)             -7.112554108976e+02
                         :        (kJ/mol)             -2.975892639195e+03
                         :            (eV)             -3.084295378071e+01
===========================================================================


===========================================================================
                         Orbital Energies (final)
---------------------------------------------------------------------------
    n  Occ Spin          :                                         Epsilon
---------------------------------------------------------------------------
    0    2    p          :            (au)                 -0.604191218847
---------------------------------------------------------------------------
 Sum occupied            :            (au)                 -1.208382437695
===========================================================================


===========================================================================
                           Dipole Moment (dip-1)
---------------------------------------------------------------------------
 r_O                     :        0.000000        0.000000        0.000000
---------------------------------------------------------------------------
 Electronic vector       :        0.000086        0.000086       -0.000003
 Magnitude               :            (au)                        0.000121
                         :         (Debye)                        0.000308
---------------------------------------------------------------------------
 Nuclear vector          :        0.000000        0.000000        0.000000
 Magnitude               :            (au)                        0.000000
                         :         (Debye)                        0.000000
---------------------------------------------------------------------------
 Total vector            :        0.000086        0.000086       -0.000003
 Magnitude               :            (au)                        0.000121
                         :         (Debye)                        0.000308
===========================================================================


                                                                           
                                                                           
***************************************************************************
***                                                                     ***
***                            Exiting MRChem                           ***
***                                                                     ***
***                       Wall time :  0h  0m  2s                       ***
***                                                                     ***
***************************************************************************
                                                                           
                                                                           
